modules/generative-aws/clients/aws.go

```diff
@@ -313,31 +313,31 @@ func (v *aws) generateForPrompt(textProperties map[string]string, prompt string)
 }
 
 func (v *aws) getAwsAccessKey(ctx context.Context) (string, error) {
-	if len(v.awsAccessKey) > 0 {
-		return v.awsAccessKey, nil
-	}
 	awsAccessKey := ctx.Value("X-Aws-Access-Key")
 	if awsAccessKeyHeader, ok := awsAccessKey.([]string); ok &&
 		len(awsAccessKeyHeader) > 0 && len(awsAccessKeyHeader[0]) > 0 {
 		return awsAccessKeyHeader[0], nil
 	}
+	if len(v.awsAccessKey) > 0 {
+		return v.awsAccessKey, nil
+	}
 	return "", errors.New("no access key found " +
-		"neither in request header: X-Aws-Access-Key " +
-		"nor in environment variable under AWS_ACCESS_KEY_ID")
+		"neither in request header: X-AWS-Access-Key " +
```

Comment:
We check on line 316 for `X-Aws-...` instead of `X-AWS-...` is it on purpose ?

Reply:
yes, it's just a cosmetic change for a better error message `X-AWS-Access-Key` looks better then `X-Aws-Access-Key` in my opinion.

One more thing to add, when you are passing headers with request then those header keys land in context in a `capitalized word format` so for example if you send a header using such a formatted keys:
 - `X-AWS-ACESS-KEY`
 - `X-AWS-Acess-Key`
 - `x-aws-access-key`
 - any other capitalization :)
 golang will pass those header keys in context always like that: `X-Aws-Acess-Key`